python/torch_mlir_e2e_test/mhlo_backends/abc.py

```diff
@@ -0,0 +1,49 @@
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+
+import abc
+from typing import TypeVar
+
+import torch
+
+from torch_mlir.ir import Module
+
+# A type shared between the result of `MhloBackend.compile` and the
+# input to `MhloBackend.load`. Each backend will likely have a
+# different definition of this type.
+CompiledArtifact = TypeVar('CompiledArtifact')
+
+# A wrapper around a backend-specific loaded program representation
+# that uniformly translates the `x.method(...)` interface expected of
+# Torch modules into appropriate lower-level operations.
+Invoker = TypeVar('Invoker')
+
+
+class MhloBackend(abc.ABC):
+    """The interface to an MHLO backend.
+
+    Backends are recommended to raise meaningful exceptions in case of error,
+    ideally with easy reproduction instructions.
+    """
+    @abc.abstractmethod
+    def compile(self, module: Module) -> CompiledArtifact:
+        """Compile the provided MLIR module into a compiled artifact.
+
+        The module adheres to the MHLO backend contract
+        (see the VerifyMhloBackendContract pass).
```

Comment:
Does the VerifyMhloBackendContract pass exist? If not, could you also add it in another PR (see VerifyTosaBackendContract). This helps to make failures clearer when debugging, which is especially important for new contributors or other project maintainers who are not familiar with MHLO.

Reply:
Thanks for reminding me. I created an issue to track it, see https://github.com/llvm/torch-mlir/issues/1267